File: task_equivalents/llm_translations/gemini/optimized/python/machine_learning_workflows/02_regression_workflow_feature_importance.py
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import GradientBoostingRegressor
from sklearn.linear_model import LassoCV
from sklearn.metrics import mean_squared_error
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import make_pipeline

Hitters = pd.read_csv('data/csv/Hitters.csv', index_col=0)
Hitters = Hitters.dropna()

drop_cols = [col for col in ['rownames', 'Player', 'Name'] if col in Hitters.columns]
if drop_cols:
    Hitters = Hitters.drop(columns=drop_cols)

Hitters['League'] = np.where(Hitters['League'] == 'A', 0, 1)
Hitters['NewLeague'] = np.where(Hitters['NewLeague'] == 'A', 0, 1)
Hitters['Division'] = np.where(Hitters['Division'] == 'E', 0, 1)
Hitters['logSal'] = np.log(Hitters['Salary'])

x = Hitters.drop(columns=['Salary', 'logSal'])
y = Hitters['logSal']

xtrain, xtest, ytrain, ytest = train_test_split(x, y, train_size=200, random_state=1)

train_mse = np.zeros(20)
lambda_grid = np.linspace(0.001, 0.201, 20)

for i, lr in enumerate(lambda_grid):
    boost_hit = GradientBoostingRegressor(loss='squared_error', n_estimators=1000, learning_rate=lr, random_state=1)
    boost_hit.fit(xtrain, ytrain)
    boost_pred = boost_hit.predict(xtrain)
    train_mse[i] = mean_squared_error(ytrain, boost_pred)

cv_out = make_pipeline(StandardScaler(), LassoCV(cv=10, random_state=1, max_iter=10000))
cv_out.fit(xtrain, ytrain)
lasso_pred = cv_out.predict(xtest)
lasso_mse = mean_squared_error(ytest, lasso_pred)

best_lambda = lambda_grid[np.argmin(train_mse)]
best_boost_hit = GradientBoostingRegressor(loss='squared_error', n_estimators=1000, learning_rate=best_lambda, random_state=1)
best_boost_hit.fit(xtrain, ytrain)

summary = pd.DataFrame({
    'var': xtrain.columns,
    'rel.inf': best_boost_hit.feature_importances_ * 100
}).sort_values(by='rel.inf', ascending=False)
print(summary)

File: task_equivalents/llm_translations/gemini/optimized/r/machine_learning_workflows/02_regression_workflow_feature_importance.R
library(gbm)
library(glmnet)

Hitters <- read.csv('data/csv/Hitters.csv')
Hitters <- na.omit(Hitters)
Hitters <- Hitters[, !(names(Hitters) %in% c('rownames', 'Player', 'Name'))]

Hitters$League <- ifelse(Hitters$League == 'N', 1, 0)
Hitters$NewLeague <- ifelse(Hitters$NewLeague == 'N', 1, 0)
Hitters$Division <- ifelse(Hitters$Division == 'W', 1, 0)
Hitters$logSal <- log(Hitters$Salary)

x <- Hitters[, !(names(Hitters) %in% c('Salary', 'logSal'))]
y <- Hitters$logSal

set.seed(1)
train_idx <- sample(seq_len(nrow(x)), 200)
xtrain <- x[train_idx, ]
xtest  <- x[-train_idx, ]
ytrain <- y[train_idx]
ytest  <- y[-train_idx]

lambda_grid <- seq(0.001, 0.201, length.out = 20)
train_mse <- numeric(20)
train_data <- cbind(xtrain, logSal = ytrain)

for (i in seq_along(lambda_grid)) {
  set.seed(2)
  boost <- gbm(logSal ~ ., data = train_data, distribution = "gaussian", 
               n.trees = 1000, shrinkage = lambda_grid[i], 
               interaction.depth = 3, bag.fraction = 1)
  
  train_mse[i] <- mean((ytrain - predict(boost, n.trees = 1000))^2)
}

set.seed(1)
lasso <- cv.glmnet(as.matrix(xtrain), ytrain, alpha = 1, nfolds = 10)
lasso_preds <- predict(lasso, s = "lambda.min", newx = as.matrix(xtest))
lasso_mse <- mean((ytest - lasso_preds)^2)

best_lambda <- lambda_grid[which.min(train_mse)]

set.seed(2)
best_boost <- gbm(logSal ~ ., data = train_data, distribution = "gaussian", 
                  n.trees = 1000, shrinkage = best_lambda, 
                  interaction.depth = 3, bag.fraction = 1)

feature_importance <- summary(best_boost, n.trees = 1000, plotit = FALSE)
print(feature_importance)